From velegrin@mail.mipt.ru  Thu Dec 18 15:56:56 2008
From: Sergey Ovcharenko <velegrin@mail.mipt.ru>
Date: Wed, 17 Dec 2008 19:28:06 +0300
Subject: USB: storage: unusual_devs.h additions for Macpower MasterBox
To: stern@rowland.harvard.edu, linux-usb@vger.kernel.org, usb-storage@lists.one-eyed-alien.net
Message-ID: <49492896.70804@mail.mipt.ru>


Jason Johnston reported these problems with his external USB hard drive:
http://bugs.gentoo.org/show_bug.cgi?id=250789

The number of reported sectors is off-by-one.

/proc/bus/usb/devices:
T:  Bus=01 Lev=02 Prnt=08 Port=00 Cnt=01 Dev#=  9 Spd=480 MxCh= 0
D:  Ver= 2.00 Cls=00(>ifc ) Sub=00 Prot=00 MxPS=64 #Cfgs=  1
P:  Vendor=0dc4 ProdID=0073 Rev= 0.00
S:  Manufacturer=Macpower Technology Co.LTD.
S:  Product=USB 2.0 3.5" DEVICE
S:  SerialNumber=03006C
C:* #Ifs= 1 Cfg#= 1 Atr=c0 MxPwr=100mA
I:* If#= 0 Alt= 0 #EPs= 2 Cls=08(stor.) Sub=06 Prot=50 Driver=usb-storage
E:  Ad=01(O) Atr=02(Bulk) MxPS= 512 Ivl=0ms
E:  Ad=82(I) Atr=02(Bulk) MxPS= 512 Ivl=0ms

Signed-off-by: Sergey Ovcharenko <velegrin@mail.mipt.ru>
Signed-off-by: Phil Dibowitz <phil@ipom.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/storage/unusual_devs.h |    7 +++++++
 1 file changed, 7 insertions(+)

--- a/drivers/usb/storage/unusual_devs.h
+++ b/drivers/usb/storage/unusual_devs.h
@@ -1537,6 +1537,13 @@ UNUSUAL_DEV(  0x0d96, 0x5200, 0x0001, 0x
 		"JD 5200 z3",
 		US_SC_DEVICE, US_PR_DEVICE, NULL, US_FL_FIX_INQUIRY),
 
+/* Reported by  Jason Johnston <killean@shaw.ca> */
+UNUSUAL_DEV(  0x0dc4, 0x0073, 0x0000, 0x0000,
+		"Macpower Technology Co.LTD.",
+		"USB 2.0 3.5\" DEVICE",
+		US_SC_DEVICE, US_PR_DEVICE, NULL,
+		US_FL_FIX_CAPACITY),
+
 /* Reported by Lubomir Blaha <tritol@trilogic.cz>
  * I _REALLY_ don't know what 3rd, 4th number and all defines mean, but this
  * works for me. Can anybody correct these values? (I able to test corrected
